src/Collection/ProcessArgumentsCollection.php

```diff
@@ -132,11 +132,24 @@ public function addOptionalBooleanArgument(string $argument, $value, string $tru
      * @param int|null $value
      */
     public function addOptionalIntegerArgument(string $argument, $value)
+    {
+        $this->addOptionalMixedArgument($argument, $value);
+    }
+
+    /**
+     * @param string $argument
+     * @param mixed $value
+     */
+    public function addOptionalMixedArgument(string $argument, $value)
     {
         if (null === $value) {
             return;
         }
 
+        if (\is_bool($value)) {
```

Comment:
Maybe a bridge to far at this point since it isn't used?
We also have `addOptionalBooleanArgument()` for this use case